ssl/t1_lib.c

```diff
@@ -3238,7 +3267,10 @@ static const tls12_hash_info tls12_md_info[] = {
     {NID_sha224, 112, EVP_sha224},
     {NID_sha256, 128, EVP_sha256},
     {NID_sha384, 192, EVP_sha384},
-    {NID_sha512, 256, EVP_sha512}
+    {NID_sha512, 256, EVP_sha512},
+    {NID_id_GostR3411_94,       128, md_gost94},
+    {NID_id_GostR3411_2012_256, 128, md_gost2012_256},
+    {NID_id_GostR3411_2012_512, 256, md_gost2012_512},
```

Comment:
This is a problem. The function tls12_get_hash_info matches the values TLSEXT_hash_* to rows in this table. TLSEXT_hash_md5 has a value 1 and corresponds to row 0, TLSEXT_hash_sha1 has a value 2 and corresponds to row 1, etc. This is fine but the GOST hashes skip a load of numbers so TLSEXT_hash_gostr3411 has a value 237, so the algorithm in tls12_get_hash_info no longer works.